flang/lib/Optimizer/CodeGen/PreCGRewrite.cpp

```diff
@@ -319,6 +320,10 @@ class CodeGenRewrite : public fir::impl::CodeGenRewriteBase<CodeGenRewrite> {
       runOn(func, func.getBody());
     for (auto global : mod.getOps<fir::GlobalOp>())
       runOn(global, global.getRegion());
+    for (auto omp : mod.getOps<mlir::omp::ReductionDeclareOp>()) {
```

Comment:
Wouldn't applying the patterns on the module directly work for all operation in it? So we could get rid of call to runOn.

Reply:
Good idea. That does indeed work